src/game/BattleGround/BattleGroundWS.h

```diff
@@ -143,6 +236,10 @@ class BattleGroundWS : public BattleGround
         void EndBattleGround(Team winner) override;
         virtual WorldSafeLocsEntry const* GetClosestGraveYard(Player* player) override;
 
+        // Flag interactions
+        void PickUpFlagFromBase(Player* source, GameObject* target);
```

Comment:
Unless I missed something, the GameObject parameter isn't used in the implementation. So that parameter can be removed.

Also, you *always* need a valid Player object to be passed to this function, because you're unconditionally dereferencing the parameter in the implementation. That's fine, but that fact is not clear with this API \(*\). A suggestion would be to make it a reference instead, because that makes the intention "a valid object is required" (aka "nullptr is not a valid value") clear even if you only look at the header.

\(*\) I admit that API in the project is often doing it like this though. We should change/improve that IMHO.